vowpalwabbit/core/include/vw/core/global_data.h

```diff
@@ -319,8 +319,8 @@ class workspace
   workspace(const VW::workspace&&) = delete;
   VW::workspace& operator=(const VW::workspace&&) = delete;
 
-  std::string get_setupfn_name(reduction_setup_fn setup);
-  void build_setupfn_name_dict(std::vector<std::tuple<std::string, reduction_setup_fn>>&);
+  std::string _get_setupfn_name(reduction_setup_fn setup);

```

Comment:
why are these functions underscored? we don't follow that practice in the code base

Reply:
The workspace is somewhat of a public API and these functions are definitely internal. The underscore implies they are implementation details. Just an idea to make intent clearer to the user

Reply:
it is a bit pythonic, if they are internal can we put them in the details namespace or make them private?

Reply:
You can't namespace a method and making them private isn't doable without making a large number of other places in the codebase friends with the workspace

Reply:
yeah but they are private member variables, the underscore is just for developer convenience

Reply:
Do you think its better to leave the names as is here? The thing I am trying to avoid is an end user relying or even using these APIs

Reply:
related to discussion: https://github.com/VowpalWabbit/vowpal_wabbit/pull/4456